Tools/clinic/clinic.py

```diff
@@ -4865,10 +4865,16 @@ def state_parameter(self, line: str | None) -> None:
             return
 
         line = line.lstrip()
-
-        if line in ('*', '/', '[', ']'):
-            self.parse_special_symbol(line)
-            return
+        func = self.function
+        match line:
+            case '*':
+                return self.parse_star(func)
+            case '[':
+                return self.parse_open_bracket(func)
+            case ']':
+                return self.parse_close_bracket(func)
+            case '/':
+                return self.parse_slash(func)
```

Comment:
My personal preference here would be to be explicit about the fact that this isn't meant to be an exhaustive match:

```suggestion
                return self.parse_slash(func)
            case _:
                pass
```

Reply:
Yeah, I though about adding such a case; let me sleep on it :)

Reply:
Alternatively, we could do something like this, and just... make it an exhaustive match! (The diff is relative to your PR branch):

```diff
-        line = line.lstrip()
         func = self.function
-        match line:
+        match line.lstrip():
             case '*':
-                return self.parse_star(func)
+                self.parse_star(func)
             case '[':
-                return self.parse_open_bracket(func)
+                self.parse_open_bracket(func)
             case ']':
-                return self.parse_close_bracket(func)
+                self.parse_close_bracket(func)
             case '/':
-                return self.parse_slash(func)
+                self.parse_slash(func)
+            case _ as line:
+                self.parse_parameter(line)
+
+    def parse_parameter(self, line: str) -> None:
+        assert self.function is not None

         match self.parameter_state:
             case ParamState.START | ParamState.REQUIRED:
```

Reply:
(Off-topic, but the fact that we have to do `assert self.function is not None` in so many methods is a bit of a code smell. If we refactored the code so that it is never set to `None` in the first place, it would probably be a more robust overall design.)

Reply:
1. I like the exhaustive match variant!
2. OT: yes, that smell is definitely there; it's time

Reply:
Exhaustive match variant added!